### ISSUE TITLE
Implement diff command redo without implicitly clearing the old one

### ISSUE THREAD

**Issue by pnedev** [pnedev/compare-plugin#80]
This still needs justification but here is the right place to discuss pros. and cons..
Thanks.

BR

**Comment by xylographe** [comment]
Hello Pavel and Yaron,

> [\[comment\]](https://github.com/pnedev/compare-plugin/issues/70#issuecomment-223301270)  I think there is no harm in making repeatable diff commands.

I'm slowly starting to lean towards that opinion. ;)
I expect to be using "Diff since Last Save" mostly in this (or similar) scenario : 

- open terminal emulator,
- `git checkout -b newbranch master`
- `npp doc.tex`
- `git checkout otherbranch doc.tex`
- switch to NPP,
- decline NPP's offer to reload `doc.tex`,
- trigger "Diff since Last Save"

While editing `doc.tex` I will be saving and re-comparing this file regularly. 
Should I hit `Alt-L` ("Diff since Last Save") by accident, an implicit "Clear Compare" would 
be harmless—easily reverted by another `git checkout otherbranch doc.tex`.

And although I don't expect to use it frequently, a re-diff with implicit clear-compare 
would be more convenient after `git checkout thirdbranch doc.tex`.

Secondly, repeating "Git/Svn diff" with implicit clear-compare would certainly be convenient.

In conclusion, re-diff would be a nice feature, but not a crucial one.

**Comment by pnedev** [comment]
Hello @xylographe ,

OK then. I'll implement it some time soon.
Thanks.

BR

**Comment by Yaron10** [comment]
Hello Pavel and  xylographe,

That's great.
Thank you.

BR

**Comment by pnedev** [comment]
Hello all,

One update on this.
While Git re-diff might be implemented in an optimal way (without clearing the current diff and deleting the temp file respectively) the other two might not be possible due to the encoding.
The reason I create and use a temp file at all is to use the N++ encoding detection when the file is opened. If I read the old file myself and simply paste the content in a new N++ doc then the encoding is not correct.
I'll do some tests but if those fail the only option is to implicitly issue "Clear Compare" and then re-diff.
And I'm not sure there is much sense in that.

BR

**Comment by Yaron10** [comment]
Hello Pavel,

Thanks for your work and for the update. I appreciate it.

BR

**Comment by pnedev** [comment]
Hello all,

I'd like just to inform you that I'll close this thread.
The diff commands redo will be difficult to implement due to the file encoding detection and is really not worth the work.
Thank you.

BR

**Comment by Yaron10** [comment]
Hello Pavel,

Again - CP is great as it is.
Thank you very much.

BR
